### PR TITLE
Call htsjdk's disableOnTheFlyModifications do disable header repairs

### LINKED ISSUE #6012: GIAB VCF import fails because htsjdk "repairs" header in vcf
Hi folks,

In evaluating Hail to see whether it fits my use case (a variant frequency database) I ran into an issue with importing VCF files from GIAB. It turns out that these use type `String` for the `PS` `##FORMAT` entry.

Subsequently, Hail fails to import these with the error:
```
is.hail.utils.HailException: HG001.vcf.gz:column 492: invalid character 'P' in integer literal
```
This is because of the default behaviour of `htsjdk` to "repair" these according to the VCF "standard".

`htsjdk` exposes `codec.disableOnTheFlyModifications` to toggle this behaviour which can be called from somewhere around https://github.com/hail-is/hail/blob/master/hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala#L1143.

Ideally I would like to expose this toggle also at the `import_vcf` method of Hail.
I'll create a PR to do so accordingly ASAP.

Comments/questions?

Thanks!

Regards,

Mark

### PR DESCRIPTION
**Pull request by marksantcroos**
Fixes #6012.

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala [hunk 1]
```diff
@@ -1138,9 +1138,12 @@ object LoadVCF {
     callFields: Set[String],
     entryFloatType: TNumeric,
     lines: Array[String],
-    arrayElementsRequired: Boolean = true
+    arrayElementsRequired: Boolean = true,
+    repairHeader: Boolean = true
   ): VCFHeaderInfo = {
     val codec = new htsjdk.variant.vcf.VCFCodec()
+    if (!repairHeader)
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: I'd be happier just turning this off in all cases without any parameterization. This has caused nothing but trouble in the past!

Comment (tpoterba) [reply hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: it certainly couldn't have been easy to find all the places we thread the parameters through, though! Nice work.

**Comment by tpoterba** [comment]
fixes #2822 🎉  🎉  🎉

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala (hunk 1 above)
Comment (marksantcroos) [reply hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: @tpoterba I'm happy to remove the conditional, if you are ok with that, that simplifies things obviously.

Comment (tpoterba) [reply hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: yes, absolutely.

**Comment by tpoterba** [comment]
Awesome, thanks for the contribution!

**Pull request merged**